src/tools/gcc.lua

```diff
@@ -318,6 +336,11 @@
 			table.insert(result, '-isystem ' .. p.quoted(dir))
 		end
 
+		for _, dir in ipairs(includedirsafter or {}) do
```

Comment:
```suggestion
		for _, dir in ipairs(includedirsafter or cfg.includedirsafter or {}) do
```
"Future" proof for generators not in core (might also apply to previous parameters too).

Reply:
I tend to disagree there. That'll change the behavior of those exporters, potentially in unexpected ways. I see this as a responsibility of the caller, not this function.

Reply:
`includedirsafter` would be ignored by unmodified generators.
With my proposal, there is (a big) chance to have expected behavior.
Else it would be broken too (in another way) and requires changes anyway (pass `cfg.includedirsafter` for your version, pass `{}` for mine, and handle `includedirsafter` another way.

codelite/gmake2 might be unmodified ;-)
xcode/visual handle it differently anyway

gmake support seems missing BTW.

Reply:
`includedirsafter` would be ignored by unmodified generators in this implementation, since arguments are optional.  Correct, XCode and Visual Studio handle it differently, as they don't support functionality like `idirafter` provides. GMake support is intentionally missing, as it's deprecated.

If we do change the behavior of how this works, I think it'd be better to have this as a separate PR, as it's out of scope of the current functionality.  As written, the new behavior matches the expected behavior from previous `includedirs` and friends.

Reply:
On top of this, if any 3rd party module passing `nil` to the function with your mechanism to attempt "skip" a parameter (rather than use `{}`) would break, as it would now check `cfg.####` before just putting in `{}`.  Given we're to beta, I'm a bit hesitant to change existing behavior.

Reply:
As usual, we disagree ;-) on expected behavior.
But ok.
(I will have to create 3 PRs to add support to the external generators ^_^)

Shouldn't doc specify that it is unsupported by gmake?
(even if deprecated, some features only works with gmake and not gmake2 (D, premake-qt, ...)
(fixing it seems simpler btw).

Reply:
I think the plan is to completely drop gmake support in 5.0 full release.  We can update the documentation to explicitly call out lack of gmake support, though.

Reply:
As long as gmake is here, either:
- Update it (add `cfg.includedirsafter` to `toolset.getincludedirs` in gmake)
- Specify in includedirsafter.md than gmake is unsupported.

First solution seems simpler/better to me.